doc/tutorial/alphabet/index.md

```diff
@@ -0,0 +1,334 @@
+\f[ c = \frac{n_G + n_C}{n_A + n_T + n_G + n_C} \f]
+Write a program that
+1. reads a sequence as command line argument into a vector of seqan3::dna5,
+2. counts the number of occurrences for each nucleotide in an array of size `alphabet size` and
+3. calculates the GC content.
+
+The seqan3::dna5 type ensures that invalid characters in the input sequence are converted to 'N'.
+Note that these characters should not influence the GC content.
+
+Pass the sequences `CATTACAG` (3/8 = 37.5%) and `ANNAGAT` (1/5 = 20%) to your program and check
+if your results are correct.
+
+\endassignment
+\solution
+\snippet alphabet_gc_content.cpp exercise
+\endsolution
+
+<br/>
+
+---
+
+# The Alphabet and NucleotideAlphabet concepts
+
+Concepts are compile-time constraints for template parameters. They are boolean predicates that limit the
+set of accepted template types or values and can therefore specialise a function or class. Let's look at an example:
+
+~~~cpp
+template <typename alphabet_type>
+alphabet_type complement(alphabet_type const alph)
+{
+    // implementation...
+}
+~~~
+
+We want to implement a function `complement` that computes the complementary nucleobase of a given one.
+Because we want a unique interface for each alphabet, e.g. seqan3::dna4 and seqan3::dna5, we implement
+it as a template function as shown above. The problem is that in this way the function does not only accept DNA
+and RNA alphabets – for which this function is useful – but also any other type, like `float` or `std::string`.
+
+The solution is to *constrain* the function to a certain set of alphabets, namely nucleotide alphabets.
+To achieve this, SeqAn defines a concept seqan3::NucleotideAlphabet that specifies how nucleotide
+alphabets look like, so the compiler can decide, whether a given type is valid or not. In our example
+we can now require that the given `alphabet_type` is a nucleotide alphabet.
+
+~~~cpp
+template <seqan3::NucleotideAlphabet alphabet_type>
+alphabet_type complement(alphabet_type const alph)
+{
+    // implementation...
+}
+~~~
+
+If you try to call the function `complement` on e.g. a `float` variable, the compilation fails with a
+helpful error message stating that `float` is not a seqan3::NucleotideAlphabet.
+
+A more general concept in SeqAn is seqan3::Alphabet. It ensures for all alphabet types in SeqAn that you can
+- get and assign the rank (seqan3::to_rank, seqan3::assign_rank)
+- get and assign the char representation (seqan3::to_char, seqan3::assign_char, seqan3::assign_char_strict)
+- check if a char is valid in this alphabet (seqan3::char_is_valid_for)
+- retrieve the alphabet size (seqan3::Alphabet::alphabet_size_v) and
+- compare their values (operators `==`, `>`, `>=`, ...).
+
+\note Of course, the seqan3::NucleotideAlphabet satisfies seqan3::Alphabet. <br/>
+      It actually extends the alphabet concept with the [complement()](\ref seqan3::NucleotideAlphabet::complement)
+      function.
+
+<br/>
+
+---
```

Comment:
Can you reduce the text of this section to talk less about concepts and more generally about what is expected of "Alphabets" or "NucleotideAlphabets"?